weblate/formats/xwiki.py

```diff
@@ -0,0 +1,100 @@
+"""Specific file formats for XWiki"""
+
+import re
+
+from django.utils.functional import cached_property
+from translate.misc import quote
+from weblate.formats.ttkit import PropertiesFormat, PropertiesUnit
+
+
+class XWikiUnit(PropertiesUnit):
+    """
+    Inspired from PropertiesUnit, allow to override the methods to use the right
+    XWikiDialect methods for decoding properties.
+    """
+
+    @cached_property
+    def source(self):
+        # Need to decode property encoded string
+        return quote.xwiki_properties_decode(super().source)
+
+    @cached_property
+    def target(self):
+        """Return target string from a Translate Toolkit unit."""
+        if self.unit is None:
+            return ""
+        # Need to decode property encoded string
+        # This is basically stolen from
+        # translate.storage.properties.propunit.gettarget
+        # which for some reason does not return translation
+        value = quote.xwiki_properties_decode(self.unit.value)
+        value = re.sub("\\\\ ", " ", value)
+        return value
+
+
+class XWikiPropertiesFormat(PropertiesFormat):
```

Comment:
Apparently that's not good for the autodetection since we are also using file name `*.properties`, so it seems better to rely on the already existing PropertiesFormat here.

Reply:
PropertiesBaseFormat has no autodetection enabled.

Reply:
I need to investigate a bit more, but apparently using it makes my tests failing right now.

Reply:
The autodetection is purely based on extension, so you don't want to have two hits here. Also it's not that relevant these days as the file format has to be configured when creating a component, so it applies only on import and in that case the format used in current component is always tried first.

Reply:
Fixed. I needed to specify the value of `new_translation` too.